assets/js/googlesitekit/datastore/user/user-info.js

```diff
@@ -397,6 +397,26 @@ export const selectors = {
 		return user !== undefined ? user.picture : user;
 	} ),
 
+	/**
+	 * Gets the url with the users email appended.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state Data store's state.
+	 * @return {(string|undefined)} The concatenated url if an email is present, undefined else.
+	 */
+	getAccountChooserURL: createRegistrySelector(
+		( select ) => ( state, destinationURL ) => {
+			invariant( destinationURL, 'destinationURL is required' );
+			const userEmail = select( CORE_USER ).getEmail();
+			return userEmail !== undefined
+				? `${ encodeURIComponent(
+						destinationURL
+				  ) }&Email=${ encodeURIComponent( userEmail ) }`
+				: undefined;
```

Comment:
Let's use a bit more spacing here to give sections of code more readability. Especially things like `invariant` calls can use some whitespace after them to separate out different logic/aspects of the function.

Also: the case-sensitive `Email` param is a bit odd, so I think it's worth adding a comment in case anyone notices it and is curious about it in the future. 🙂 

Oh, and one last thing: let's try to avoid ternary operators whenever possible. I think they tend to make code a lot less readable, and in this case we can do a return early rather than ternary that makes the logic/intent clearer:

```suggestion
			invariant( destinationURL, 'destinationURL is required' );

			const userEmail = select( CORE_USER ).getEmail();
			
			if ( userEmail !== undefined ) {
				return undefined;
			}

			// The `Email` query parameter is case-sensitive; a capital "E"
			// is required for the account chooser URL.
			return ` https://accounts.google.com/accountchooser?continue=${ encodeURIComponent(
				destinationURL
			) }&Email=${ encodeURIComponent( userEmail ) }`;
```